crates/fuel-core/src/database/sealed_block.rs

```diff
@@ -127,12 +127,21 @@ impl Database {
         }
     }
 
-    pub fn get_transactions_on_block(
+    pub fn get_transactions_on_blocks(
         &self,
-        block_id: &BlockId,
-    ) -> StorageResult<Option<Vec<Transaction>>> {
-        Ok(self
-            .get_sealed_block_by_id(block_id)?
-            .map(|Sealed { entity: block, .. }| block.into_inner().1))
+        block_height_range: Range<u32>,
+    ) -> StorageResult<Option<Vec<Transactions>>> {
+        let transactions = block_height_range
+            .into_iter()
+            .map(BlockHeight::from)
+            .map(|block_height| {
+                let transactions = self
+                    .get_sealed_block_by_height(&block_height)?
```

Comment:
Is this the right name still? I feel like `get_transactions_by_block_height` would be more accurate.